dialect/sql/sqlgraph/graph_test.go

```diff
@@ -805,6 +805,35 @@ WHERE "s1"."users"."id" IN
 	}
 }
 
+func TestHasNeighborsWithContext(t *testing.T) {
+	type key string
+	ctx := context.WithValue(context.Background(), key("mykey"), "myval")
+	for _, rel := range [...]Rel{M2M, O2M, O2O} {
+		t.Run(rel.String(), func(t *testing.T) {
+			sel := sql.Dialect("postgres").
+				Select("*").
+				From(sql.Table("users")).
+				WithContext(ctx)
+			step := NewStep(
+				From("users", "id"),
+				To("groups", "id"),
+				Edge(rel, false, "user_groups", "user_id", "group_id"),
+			)
+			var called bool
+			pred := func(s *sql.Selector) {
+				called = true
+				if got := s.Context().Value(key("mykey")).(string); got != "myval" {
+					t.Fatalf("expected selector context to have %q but got %q", "myval", got)
+				}
+			}
+			HasNeighborsWith(sel, step, pred)
+			if !called {
+				t.Fatal("expected predicate function to be called")
+			}
```

Comment:
```suggestion
			pred := func(s *sql.Selector) {
				called = true
				got := s.Context().Value(key("mykey")).(string)
				require.Equal(t,  "myval" , got)
			}
			HasNeighborsWith(sel, step, pred)
			require.True(t, called, "expected predicate function to be called")
```

Minor comment - I'm not fan of the `testify/require` package, but we already use it, so let's stay consistent with the existing codebase. Thanks ❤️